### PR TITLE
If 299 create new applet

### PR DESCRIPTION
**Pull request by seemueller**

### CODE REVIEW

**Comment by mattmikulina** [comment]
Task linked: [IF-299 Create new Applet](https://app.clickup.com/t/2177764/IF-299)

File: fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-conditions/applet-conditions.component.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+
+  .operator-dropdown{
+    padding: 16px;
+    border-radius: 20px;
+    opacity: 1;
+    box-shadow: 0 3px 6px #00000029;
```
Comment (jsylupp) [inline fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-conditions/applet-conditions.component.scss:28]: use variable

File: fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-conditions/applet-conditions.component.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  @Input()
+  ruleGroup: FormGroup;
+  conditionsGroup: FormGroup;
+  devices: Device[];
+  dropdownOperatorOptions: SelectItem[];
+
+  constructor(private oispService: OispService,
+              private formBuilder: FormBuilder,
+              private enumHelpers: EnumHelpers) {
+    this.dropdownOperatorOptions = this.getOperatorDropdownValue();
```
Comment (jsylupp) [inline fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-conditions/applet-conditions.component.ts:50]: rename suggestion: consider using wording options also in the method

File: fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-conditions/applet-conditions.component.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    this.loadDevices();
+    this.createFormgroups();
+  }
+
+  ngOnInit(): void {
+    if (!this.conditions.operator) {
```
Comment (jsylupp) [inline fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-conditions/applet-conditions.component.ts:56]: consider extracting in method(s) due to faster understanding of purpose in name

File: fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-conditions/applet-conditions-value/condition-time-selector/condition-time-selector.component.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+export class ConditionTimeSelectorComponent implements OnInit {
+
+  private readonly SECOND = 1;
+  private readonly MINUTE = 60 * this.SECOND;
+  private readonly HOUR = 60 * this.MINUTE;
+  private readonly DAY = 24 * this.HOUR;
+  private readonly WEEK = 7 * this.DAY;
+  private readonly TIME_OPERATORS = [this.SECOND, this.MINUTE, this.HOUR, this.DAY, this.WEEK];
+
+  conditionTimeDropdownValue: SelectItem[];
```
Comment (jsylupp) [inline fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-conditions/applet-conditions-value/condition-time-selector/condition-time-selector.component.ts:34]: maybe "...Options" to have consistent naming

File: fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-conditions/applet-conditions-value/condition-time-selector/condition-time-selector.component.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    let timeOperator: number;
+    for (const item of this.TIME_OPERATORS) {
+      if (value % item === 0) {
+        timeOperator = item;
+      }
+    }
+    this.timeGroup.patchValue({ timeOperator, value: value / timeOperator});
+  }
+
+  private getConditionTimeDropValue() {
```
Comment (jsylupp) [inline fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-conditions/applet-conditions-value/condition-time-selector/condition-time-selector.component.ts:69]: maybe rename to ..Options?

File: fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-action-list/applet-action/applet-action.component.scss [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  .email-list {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+
+  .list-item {
+    background: #FFFFFF 0% 0% no-repeat padding-box;
```
Comment (jsylupp) [inline fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-action-list/applet-action/applet-action.component.scss:53]: use variable

File: fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-action-list/applet-action/applet-action.component.scss [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    padding: .8rem;
+    margin-right: 5px;
+  }
+
+  .full-width {
+    width: 100%;
+  }
+
+  .add-button {
+    background: #FFFFFF 0% 0% no-repeat padding-box;
```
Comment (jsylupp) [inline fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-action-list/applet-action/applet-action.component.scss:68]: use variable

File: fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-action-list/applet-action/applet-action.component.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+  ngOnInit(): void {
+    if ((this.actionGroup.get('target') as FormArray).length === 0) {
+      (this.actionGroup.get('target') as FormArray).push(
+        new FormControl(null, [Validators.required, Validators.minLength(1)])
+      );
+    }
+  }
+
+  showActionTypePanel(type: RuleActionType): boolean {
```
Comment (jsylupp) [inline fusionfrontend/src/app/fusion-applets/components/fusion-applet-editor/applet-action-list/applet-action/applet-action.component.ts:43]: Name könnte evtl verwirren, da bei show bisher meist etwas angezeigt wurde. Vlt is...Visible

**Review by jsylupp — CHANGES_REQUESTED** [review]

File: fusionfrontend/src/app/fusion-applets/components/fusion-applets-overview/fusion-applets-overview.component.ts [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    if (this.isRouteActive('overview')) {
+      return rules.filter(rule => !archivStatus.includes(rule.status) );
+    } else {
+      return rules.filter(rule =>  archivStatus.includes(rule.status) );
+    }
+  }
+
+  isRouteActive(subroute: string): boolean {
+    const snapshot = this.activatedRoute.snapshot;
+    return snapshot.url.map(sement => sement.path).includes(subroute);
```
Comment (jsylupp) [inline fusionfrontend/src/app/fusion-applets/components/fusion-applets-overview/fusion-applets-overview.component.ts:64]: do you mean segment?

**Pull request merged**